cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/batch/BasicMapReduceContext.java

```diff
@@ -223,6 +228,10 @@ public void addOutput(Output output) {
                                              "Add the output as a DatasetOutput.");
       }
       providedOutput = new ProvidedOutput(output, outputFormatProvider);
+    } else if (output.getClass().getCanonicalName().startsWith(CDAP_PACKAGE_PREFIX)) {
+      // Skip unsupported outputs from within CDAP packages.
+      // This is used to ignore unsupported outputs in MapReduce (such as the SQL Engine Output for Spark).
```

Comment:
should it check for the SQLEngineOutput only for now or do you feel we will have other outputs sometime soon?

Reply:
We won't have any other outputs anytime soon. 

I don't want to add an external dependency to the `cdap-etl-api` module just to check if this is the package I care about.

I also don't want to be hardcoding all package names of all outputs I may create in the future.

Reply:
Thanks.